process/block/preprocess/transactionsV2.go

```diff
@@ -478,14 +678,64 @@ func (txs *transactions) displayProcessingResults(
 			"sc txs", mapScsForShard[miniBlock.ReceiverShardID])
 	}
 
-	log.Debug("createAndProcessMiniBlocksFromMeV2 has been finished",
-		"total txs", nbSortedTxs,
+	log.Debug("transactions info", "total txs", nbSortedTxs,
 		"num txs added", processingInfo.numTxsAdded,
-		"num txs bad", processingInfo.numTxsBad,
+		"num bad txs", processingInfo.numBadTxs,
 		"num txs failed", processingInfo.numTxsFailed,
 		"num txs skipped", processingInfo.numTxsSkipped,
 		"num txs with initial balance consumed", processingInfo.numTxsWithInitialBalanceConsumed,
 		"num cross shard sc calls or special txs", processingInfo.numCrossShardScCallsOrSpecialTxs,
 		"used time for computeGasConsumed", processingInfo.totalTimeUsedForComputeGasConsumed,
-		"used time for processAndRemoveBadTransaction", processingInfo.totalTimeUsedForProcess)
+		"used time for processAndRemoveBadTransaction", processingInfo.totalTimeUsedForProcess,
+	)
+}
+
+func (txs *transactions) displayProcessingResultsOfScheduledMiniBlocks(
+	totalGasConsumedInSelfShard uint64,
+	mapGasConsumedByMiniBlockInReceiverShard map[uint32]map[TxType]uint64,
+	miniBlocks block.MiniBlockSlice,
+	schedulingInfo *scheduledTxsInfo,
+	nbSortedTxs int,
+) {
+	if len(miniBlocks) > 0 {
+		log.Debug("scheduled mini blocks from me created", "num", len(miniBlocks))
+	}
+
+	log.Debug("displayProcessingResultsOfScheduledMiniBlocks",
+		"self shard", txs.shardCoordinator.SelfId(),
+		"total gas consumed in self shard", totalGasConsumedInSelfShard)
+
+	for _, miniBlock := range miniBlocks {
+		log.Debug("scheduled mini block info",
+			"type", "ScheduledBlock",
+			"sender shard", miniBlock.SenderShardID,
+			"receiver shard", miniBlock.ReceiverShardID,
+			"gas consumed in receiver shard", mapGasConsumedByMiniBlockInReceiverShard[miniBlock.ReceiverShardID][scTx],
+			"txs added", len(miniBlock.TxHashes))
+	}
+
+	log.Debug("scheduled transactions info", "total txs", nbSortedTxs,
+		"num scheduled txs added", schedulingInfo.numScheduledTxsAdded,
+		"num scheduled bad txs", schedulingInfo.numScheduledBadTxs,
+		"num scheduled txs skipped", schedulingInfo.numScheduledTxsSkipped,
+		"num scheduled txs with initial balance consumed", schedulingInfo.numScheduledTxsWithInitialBalanceConsumed,
+		"num scheduled cross shard sc calls", schedulingInfo.numScheduledCrossShardScCalls,
+		"used time for scheduled computeGasConsumed", schedulingInfo.totalTimeUsedForScheduledComputeGasConsumed,
+		"used time for scheduled VerifyTransaction", schedulingInfo.totalTimeUsedForScheduledVerify,
+	)
+}
+
+func getAdditionalTimeFunc() func() bool {
+	startTime := time.Now()
+	return func() bool {
+		return additionalTimeForCreatingScheduledMiniBlocks > time.Since(startTime)
+	}
+}
+
+// isScheduledMiniBlock returns if the given mini block is of type scheduled or not
+func isScheduledMiniBlock(miniBlock *block.MiniBlock) bool {
```

Comment:
you could implement this at the miniblock level as a public function.

Reply:
done